src/test/java/com/wikia/webdriver/testcases/mercurytests/SearchTests.java

```diff
@@ -11,42 +11,108 @@
-  public void mercury_search_navigateToPageUsingSearch() {
-    String searchingPhrase = "Infobox";
+  @Test(groups = "mercury_search_navigateToPageUsingSearchSuggestions")
+  public void mercury_search_navigateToPageUsingSearchSuggestions() {
+    init();
+
+    try {
+      this.topBar.openSearch().typeInSearch(SEARCH_PHRASE).selectSearchSuggestion(0);
+    } catch (Exception e) {
+      System.out.println(e);
+    }
+
+    Assertion.assertTrue(driver.getCurrentUrl().contains(SEARCH_PHRASE));
+  }
+
+  @Test(groups = "mercury_search_navigateToPageUsingSearchResults")
+  public void mercury_search_navigateToPageUsingSearchResults() {
     init();
 
-    topBar.openSearch().typeInSearch(searchingPhrase).selectSearchSuggestion(0);
+    SearchResultsPage searchResults = new SearchResultsPage().openForQuery(SEARCH_PHRASE);
+    searchResults.selectSearchResult(0);
 
-    Assertion.assertTrue(driver.getCurrentUrl().contains(searchingPhrase));
+    Assertion.assertTrue(driver.getCurrentUrl().contains(SEARCH_PHRASE));
   }
 
   @Test(groups = "mercury_search_cancelSearchPhrase")
   public void mercury_search_clearSearchPhrase() {
-    String searchingPhrase = "Infobox";
     init();
 
-    Search search = topBar.openSearch();
-    search.typeInSearch(searchingPhrase);
-    search.clickClearSearchButton();
+    this.topBar.openSearch().typeInSearch(SEARCH_PHRASE).clickClearSearchButton();
+
+    Assertion.assertEquals(this.getSearch().getSearchPhrase(), "");
+  }
+
+  @Test(groups = "mercury_search_verifySearchLayout")
+  public void mercury_search_verifySearchLayout() {
+    init();
+
+    Search searchInstance = this.topBar.openSearch();
+
+    Assertion.assertTrue(searchInstance.isSearchInputFieldVisible());
+    Assertion.assertTrue(searchInstance.isClearSearchButtonVisible());
+    Assertion.assertTrue(searchInstance.isInputFieldSearchIconVisible());
+  }
+
+  @Test(groups = "mercury_search_userIsRedirectedToSearchResultsPage")
+  public void mercury_search_userIsRedirectedToSearchResultsPage() {
+    init();
+
+    SearchResultsPage searchResults =
+        this.topBar.openSearch().typeInSearch(SEARCH_PHRASE)
+            .clickEnterAndNavigateToSearchResults();
+    Assertion.assertTrue(searchResults.isSearchResultsPageOpen());
+  }
+
+  @Test(groups = "mercury_search_searchResultsPageHasNoSearchIconInTopBar")
+  public void mercury_search_searchResultsPageHasNoSearchIconInTopBar() {
+    init();
+
+    new SearchResultsPage().openForQuery(SEARCH_PHRASE);
+    Assertion.assertFalse(this.topBar.isSearchIconClickable());
+  }
+
+  @Test(groups = "mercury_search_searchInputDoesNotCoverNavigation",
+      expectedExceptions = WebDriverException.class)
+  public void mercury_search_searchInputDoesNotCoverNavigation() {
+    init();
+
+    SearchResultsPage searchResults = new SearchResultsPage().openForQuery(SEARCH_PHRASE);
+    this.topBar.openNavigation();
+    searchResults.typeInSearch(SEARCH_PHRASE);
```

Comment:
I'm worried that this might be misleading to someone who is trying to understand what is the tested scenario. What do you think about adding an Assertion and boolean method for validate if user can type in the Input field? Of course this is just an example.